<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-10-21" />

<title>Spawning</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Mackerel Egg production</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="0.0_settings.html">
    <span class="fa fa-cogs"></span>
     
  </a>
</li>
<li>
  <a href="1.0_read.html">Read</a>
</li>
<li>
  <a href="2.0_DEP.html">DEP</a>
</li>
<li>
  <a href="3.0_Spawning.html">Spawning</a>
</li>
<li>
  <a href="4.0_TEP.html">TEP</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Spawning</h1>
<h4 class="date">2024-10-21</h4>

</div>


<pre><code># SET-UP</code></pre>
<pre><code>## Warning: le package &#39;knitr&#39; a été compilé avec la version R 4.3.3</code></pre>
<pre class="r"><code>##### my packages ################################################################################
## CRAN
cran.packages &lt;- c(&#39;tidyverse&#39;,&#39;boot&#39;,&#39;magrittr&#39;,&#39;ggpmisc&#39;,&#39;ggpubr&#39;,&#39;ggthemes&#39;,&#39;mgcv&#39;,
                   &#39;fields&#39;, &#39;inlabru&#39;, &#39;sf&#39;, &#39;PresenceAbsence&#39;, &#39;verification&#39;, &#39;raster&#39;,
                   &#39;scales&#39;, &#39;nlme&#39;,&#39;nls.multstart&#39;, &#39;stringr&#39;, &#39;ggforce&#39;)



install.this &lt;- cran.packages[!(cran.packages %in% utils::installed.packages()[,&quot;Package&quot;])]
if(length(install.this)&gt;=1) install.packages(install.this)
dummy &lt;- lapply(cran.packages, require, character.only = TRUE)

## github
git.packages &lt;- c(&#39;catchR&#39;,&#39;DFOdata&#39;,&#39;CCAM&#39;, &#39;INLA&#39;)
install.this &lt;- git.packages[!(git.packages %in% utils::installed.packages()[,&quot;Package&quot;])]
if(&#39;catchR&#39; %in% install.this)  devtools::install_github(&quot;iml-assess/catchR@eli_parallel&quot;)
if(&#39;DFOdata&#39; %in% install.this)  devtools::install_github(&quot;im-assess/DFOdata&quot;)
if(&#39;CCAM&#39; %in% install.this)  devtools::install_github(&quot;elisvb/CCAM&quot;)
if(&#39;INLA&#39; %in% install.this)install.packages(&quot;INLA&quot;,repos=c(getOption(&quot;repos&quot;),INLA=&quot;https://inla.r-inla-download.org/R/stable&quot;), dep=TRUE)

dummy &lt;- lapply(git.packages, require, character.only = TRUE)

##### source R directory  ############################################################################
#invisible(sapply(list.files(pattern=&quot;[.]R$&quot;, path=&quot;R/&quot;, full.names=TRUE), source))

##### my ggplot theme ################################################################################
theme_set(theme_mackerel())             # theme_mackerel from catchR
update_geom_defaults(&quot;line&quot;, list(size = 1))  # no idea why somethimes I get fat lines otherwise

##### passwords databases #############################################################################
source(&quot;bdOracle.R&quot;)

source(paste0(&quot;R/&quot;,my.year,&quot;/utils/basemap.R&quot;)) #if error &quot;plot new has not been call&quot;, restart R, package compatibility issues
source(paste0(&quot;R/&quot;,my.year,&quot;/utils/mackerel_fun_incubation.R&quot;)) # Mackerel incubation .
source(paste0(&quot;R/&quot;,my.year,&quot;/utils/spatial_projections.R&quot;))#
source(paste0(&quot;R/&quot;,my.year,&quot;/utils/extract_biochem.R&quot;))
source(paste0(&quot;R/&quot;,my.year,&quot;/utils/extract_T0_10.R&quot;))
source(paste0(&quot;R/&quot;,my.year,&quot;/INLA/Mesh.R&quot;))
source(paste0(&quot;R/&quot;,my.year,&quot;/INLA/INLA_ZAG_covar_Stations.R&quot;))
source(paste0(&quot;R/&quot;,my.year,&quot;/INLA/INLA_ZAG_covar_Stations_CV.R&quot;))
source(paste0(&quot;R/&quot;,my.year,&quot;/utils/nlme_boot.R&quot;))
source(paste0(&quot;R/&quot;,my.year,&quot;/INLA/INLA_tw_covar_Stations.R&quot;))
source(paste0(&quot;R/&quot;,my.year,&quot;/INLA/INLA_tw_covar_Stations_CV.R&quot;))



source(paste0(&#39;R/&#39;,my.year,&#39;/INLA/plotSpatialFieldCL.R&#39;))
source(paste0(&#39;R/&#39;,my.year,&#39;/INLA/plotSmoother.R&#39;))
source(paste0(&#39;R/&#39;,my.year,&#39;/INLA/model_validation.R&#39;))
source(paste0(&#39;R/&#39;,my.year,&#39;/INLA/get_prediction_grid.R&#39;))
source(paste0(&#39;R/biochem/PL_Get_SampleID_Batch.R&#39;))
source(paste0(&#39;R/biochem/PL_Get_Counts_Batch.R&#39;))
source(paste0(&#39;R/biochem/PL_Read_Filter.R&#39;))
source(paste0(&#39;R/biochem/PL_Taxonomic_Grouping.R&#39;))


log10p1_trans = function() scales::trans_new(&quot;log10p1&quot;, transform=function(x) log10(x+1), inverse=function(x) (10^x)-1)#inverse function is necessary for legend

#source(paste0(&quot;R/&quot;,year_to_report,&quot;/INLA/getvar.R&quot;)) # needs to be retought</code></pre>
<pre class="r"><code>new=T
N=100
load(file=paste0(&quot;data/&quot;,my.year,&quot;/DEP.RData&quot;))
load(file=paste0(&quot;data/&quot;,my.year,&quot;/eggt1.RData&quot;))
dat=eggt1</code></pre>
<div id="read" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Read</h1>
<pre class="r"><code>#prop_spawn(year,year=year, dat = egg, cv = T) # dat is used to calculate median date and in split with trajet.

date.med &lt;- eggt1 %&gt;%  filter(year !=1979) %&gt;%
  ungroup() %&gt;%
  group_by(year) %&gt;%
  dplyr::summarize(
    date.med = round(median(doy, na.rm = T)),
    mindoy = min(doy, na.rm = T),
    maxdoy = max(doy, na.rm = T)
  )

date.med &lt;- date.med %&gt;% mutate(date.med = ifelse(year == 1979, 166, date.med)) # from Ouellet et al. 1987
date.med &lt;- as.data.frame(date.med)

bio &lt;- get.bio(species = &quot;maquereau&quot;, user = my.env$bio.username, password = my.env$bio.password)

    
      bio &lt;- bio %&gt;% mutate(doy = as.numeric(format(as.Date(paste(year, month, day, sep = &quot;/&quot;), format = &quot;%Y/%m/%d&quot;), format = &quot;%j&quot;))) %&gt;%
        dplyr::filter(
          !is.na(weight), !is.na(wgonad), !is.na(doy),
          weight &gt; 0, weight &gt; wgonad, matur != 0, nafo %in% c(&quot;4T&quot;, &quot;4V&quot;, &quot;4W&quot;)
        ) 
    
      
bio %&lt;&gt;% mutate(gsi = (wgonad / weight) * 100)

#remove outliers
df &lt;- bio %&gt;%
  dplyr::filter(!is.na(gsi), matur &gt; 4, year &gt; 1978, !(gsi &gt; 5 &amp; doy &gt; 250), !(gsi &gt; 10 &amp; doy &gt; 200 &amp; year %in% c(1994, 1999, 2018, 2019, 2020)), !(year == 1999 &amp; matur &gt; 6 &amp; nafo == &quot;4V&quot;), !(nafo!=&quot;4T&quot; &amp; matur &gt; 6), !(nafo!=&quot;4T&quot; &amp; matur&gt;6 &amp; month &gt;7), length.frozen &gt; 0.2, weight &gt; 0.03, gsi &lt; 40)  %&gt;%  
  group_by(agef) %&gt;% filter(weight&gt;quantile(weight,0.0001), weight&lt;quantile(weight,0.9999))%&gt;% 
  arrange(year)


df &lt;- df %&gt;%
  arrange(year) %&gt;%
  mutate(year = as.numeric(as.character(year))) %&gt;%  filter(!is.na(gsi), !is.na(doy), !is.na(year)) %&gt;% mutate(maturcat = ifelse(!is.na(matur) &amp; matur == 6, &quot;6&quot;,                                                                                                                                 ifelse(is.na(matur), &quot;other&quot;, &quot;other&quot;))) %&gt;%  ungroup()



df&lt;- df %&gt;%  mutate(gsi=gsi/100)</code></pre>
<div id="checks" class="section level2 tabset" number="1.1">
<h2 class="tabset"><span class="header-section-number">1.1</span>
CHECKS</h2>
<div id="check-for-outliers" class="section level3" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> Check for
outliers</h3>
<pre class="r"><code>df %&gt;% filter(year %in% year_to_plot) %&gt;% ggplot(aes(x=doy, y=gsi)) +geom_point(aes(col=maturcat)) + facet_wrap(~year) + scale_color_manual(values = c(&quot;green3&quot;, &quot;grey&quot;)) </code></pre>
<p><img src="3.0_Spawning_test_GSI_proportions_files/figure-html/check%20outliers-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="model" class="section level3 tabset" number="1.1.2">
<h3 class="tabset"><span class="header-section-number">1.1.2</span>
Model</h3>
<pre class="r"><code>
#evaluate starting values

if(new){
start.val &lt;- nls.multstart::nls_multstart(gsi ~ SSlogis(doy, Asym, xmid, scal),
                                          data = df,
                                          start_lower = c(Asym =0, xmid = min(df$doy), scal = -100),
                                          start_upper = c(Asym = 1, xmid = max(df$doy), scal = 100),
                                          iter = 500,
                                          supp_errors = &quot;Y&quot;
)


nlme.fixAR &lt;- nlme(gsi ~ SSlogis(doy, Asym, xmid, scal),
  data = df,
  fixed = Asym + xmid + scal ~ 1,
  start = coefficients(start.val),
  correlation = corAR1(),
  random = Asym + xmid + scal ~ 1 | year
)

nlme_coefAR = as_tibble(coef(nlme.fixAR), rownames = &#39;year&#39;)

#keep starting values in case of needed for update
#save(nlme.fixAR,start.val,df, nlme_coefAR, file = paste0(&quot;results/&quot;, my.year,&quot;/spawning/nlmeAR&quot;, my.year,&quot;.RData&quot;))
}# else{#load(paste0(&quot;results/&quot;, my.year,&quot;/spawning/nlmeAR&quot;, my.year,&quot;.RData&quot;))}

predtimes &lt;- expand_grid(year = unique(df$year), doy = seq(0.5, 364.5)) # doy aligned to calculate probability at doy


nlme.fixpredsAR1 &lt;- predtimes %&gt;%
  mutate(fitted =predict(nlme.fixAR, newdata = predtimes,interval = &#39;confidence&#39;)) %&gt;%
  group_by(year) %&gt;%
  mutate(
    slope = c(diff(fitted, lag = 1), NA),
    prob = slope / sum(slope, na.rm = T)
  ) %&gt;%
  filter(!is.na(prob)) %&gt;%
  mutate(
    probcum0025 = cumsum(prob),
    #probcum9725 = rev(cumsum(rev(prob))),
    doy = ceiling(doy)
  ) # get back the real doy</code></pre>
<div id="check" class="section level4" number="1.1.2.1">
<h4><span class="header-section-number">1.1.2.1</span> Check</h4>
<pre class="r"><code>valid&lt;- data.frame(resid=residuals(nlme.fixAR , type=&quot;pearson&quot;), mu=fitted(nlme.fixAR), obs=df$gsi, year=df$year)

h&lt;- ggplot(valid, aes(x=mu, y=resid))+geom_point(col=&quot;grey35&quot;, size=0.5) +theme_few() + geom_hline(yintercept=0, col=&quot;red&quot;, lty=2)
f&lt;- ggplot(valid, aes(x=df$gsi, y=mu))+geom_point(col=&quot;grey35&quot;, size=0.5)+theme_few() +geom_abline(slope=1, intercept=0, col=&quot;red&quot;, lty=2)+ geom_smooth(method=&quot;lm&quot;)

hEN &lt;-  h+ ylab(&quot;Pearson&#39;s residuals&quot;) + xlab(&quot;Fitted values&quot;)

fEN &lt;-  f+ ylab(&quot;Fitted values&quot;) + xlab(&quot;Observed values&quot;)

ggarrange(hEN, fEN, ncol=2)</code></pre>
<p><img src="3.0_Spawning_test_GSI_proportions_files/figure-html/spawvalid1-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>

h&lt;- ggplot(valid, aes(x=mu, y=resid))+geom_point(col=&quot;grey35&quot;, size=0.5) +theme_few() + facet_wrap( ~ year) +geom_hline(yintercept=0, col=&quot;red&quot;, lty=2)
f&lt;- ggplot(valid, aes(x=df$gsi, y=mu))+geom_point(col=&quot;grey35&quot;, size=0.5)+theme_few() + facet_wrap( ~ year) +geom_abline(slope=1, intercept=0, col=&quot;red&quot;, lty=2)+ geom_smooth(method=&quot;lm&quot;)

hEN &lt;-  h+ ylab(&quot;Pearson&#39;s residuals&quot;) + xlab(&quot;Fitted values&quot;)

ggarrange(hEN, fEN, ncol=2)</code></pre>
<p><img src="3.0_Spawning_test_GSI_proportions_files/figure-html/spawvalid1-2.png" width="864" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
